docs: Add information for configuring highlighting (#32726)
sourcegraph/sourcegraph: doc/dev/how-to/add_support_for_a_language.md

## Syntax highlighting support

> ### Customizing Syntax Highlight Language

Reviewer: ```suggestion
### Customizing syntax highlight language
```

(Matching case convention for other headers.)

The following settings apply only to the site settings. They are global configuration options for your Sourcegraph instance.

> If you have a custom language that is derived from an existing language, it is possible to configure Sourcegraph to highlight that language as another.

Reviewer: ```suggestion
If you have a custom language based on another language that can be highlighted by Sourcegraph, it is possible to configure Sourcegraph to highlight the custom language as the supported language.
```

For example:

```json
{
  "syntaxHighlighting": {
   "languages": {
     "extensions": {
        "strato": "scala"
      },
      "patterns": []
    }
  }
}
```

> If you have custom file extensions that map to an existing language, it is possible to configure Sourcegraph to highlight those files as an existing language.

Reviewer: ```suggestion
If you have custom file extensions that map to a supported language, it is possible to configure Sourcegraph to highlight those files as the supported language.
```

For example:

```json
{
  "syntaxHighlighting": {
   "languages": {
     "extensions": {
        "module": "php",

>         "inc": "php"

Reviewer: What does `"inc"` mean?

      },
      "patterns": []
    }
  }
}
```

NOTE: In both cases, the `.` is dropped from the file extension.

> Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the syntax highlight language for that file.

Reviewer: ```suggestion
Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the language used for highlighting that file.
```

> Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the syntax highlight language for that file.

Reviewer: ```suggestion
Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the language used for highlighting that file.
```

For example:

```json
{
  "syntaxHighlighting": {
    "languages": {
      "extensions": {},
      "patterns": [
        {
          "language": "bash",
          "pattern": "bash.rc"
        },
        {
          "language": "bash",
          "pattern": ".bashprofile"
        }
      ]
    }
  }
}
```

> ### Adding New Syntax Highlighting

Reviewer: ```suggestion
### Adding new syntax highlighters
```
